---
title: "Abstract Isabel Rivera Plata"
author: "Isabel Rivera Plata with Roger Day"
date: "7/8/2019"
output:
  word_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)

options(width=132)
```


## The introduction of genomic biomarkers for breast cancer

```{r googlesheets}
library(googlesheets)
my_gs_meta = gs_title('Gene biomarker survey')
my_gs = gs_read(my_gs_meta, ws = 1) [-1, ]
my_gs = data.frame(ROW=1:nrow(my_gs), my_gs)
#  Almost -- some minor but critical serror.
```

```{r}
###  CAUTION:
###   FIRST do this in the RStudio console:
###  save.image(".RData")  ---   Do NOT run this line in this document!

attach('.RData')   ### Making sure that previously created objects are available.

library(readxl)
gbs_all = readxl::read_excel('Gene biomarker survey.xlsx')
gbs_all$ROW = 1:nrow(gbs_all)
gbs_all = gbs_all[c(ncol(gbs_all), 1:(ncol(gbs_all)-1))]
gbs = gbs_all[!is.na(gbs_all$STATUS) & (gbs_all$STATUS == 'DONE'),  ]
```

### BACKGROUND

Biomarkers are a way in which we measure some aspects of disease to support clinical decisions. In the early 2000's, two gene expression-based biomarker tests, Oncotype DX and Mammaprint, were developed as prognostic assays for breast cancer with low risk of recurrence, to reduce unnecessary overtreatment with chemotherapy. We traced and compared the development of these two tests through the years using the research literature. Our main features of interest were the focus questions, study types, and effect measures. 


### METHODS
We automated the extraction of `r length(titles)` relevant articles from PubMed into R via XML, did a stratified random selection of `r length(pmidSample)` articles, transferred their PubMed identifiers into a Google sheet, developed a data dictionary. Using the sheet we extracted the article features of interest manually. After manual extraction we summarized the patterns across time and the differences between the two tests. 


### RESULTS

The distribution of all the articles across time is :

```{r}
testString = paste(mammaprint_in_TiAbKw,oncotype_in_TiAbKw)
testString [ testString == "FALSE TRUE"]= "ODX"
testString [ testString == "TRUE TRUE"]= "both"
testString [ testString == "TRUE FALSE"]= "MP"
table(testString[], years[])

```

At this time the number of articles that the team has summarized is 

```{r}
oneswehavedone = match(x = gbs$PMID, table = pmid) 
table(testString[ oneswehavedone  ],
      years[ oneswehavedone  ] )
  
```

The first article introducing each test was 2002 for Mammaprint and 2004 for Oncotype DX. However, there was an explosion of articles on Oncotype DX in subsequent years. CHECK:  That explosion was not due to an accuulation of questions and answers about patients' clinical outcomes, but primarily focused on considerations of economic, usage of the test, effects on clinical decisions, confidence of doctor and patient, and similar directions. Mammaprint arose in Europe (Netherlands); differences in health systems may account for the focus on economics.
```{r}
my_gs$year = years[match(my_gs$PMID, pmid)]
my_gs$title = titles[match(my_gs$PMID, pmid)]
my_gs$abstract = abstracts[match(my_gs$PMID, pmid)]

table( (1:nrow(my_gs)) %in% 
    grep('Economics|cost', ignore.case = TRUE, perl = TRUE, my_gs$abstract),
    my_gs$FOCUS=='Economics')
economics_mismatch = my_gs$PMID[
  which( (
    (1:nrow(my_gs)) %in% 
      grep('Economics|cost', ignore.case = TRUE, perl = TRUE, my_gs$abstract)
  )
  !=
    (my_gs$FOCUS=='Economics')
  ) 
  ]
my_gs$abstract[my_gs$PMID %in% economics_mismatch]
(my_gs$FOCUS=='Economics')[my_gs$PMID %in% economics_mismatch]
### We are correct for 
### The mismatch that we called economics was a Genomic Health publicity blurb.

```
We detected several examples where our personal judgments classifications were correct in which an automated "bag-of-words" method would be incorrect.
```{r}
gs_done = my_gs[!is.na(my_gs$STATUS) & (my_gs$STATUS=='DONE') , ]
table(gs_done$FOCUS)
table(gs_done$year, exclude=NULL)


glm(data=my_gs, subset = !is.na(my_gs$STATUS) & (my_gs$STATUS=='DONE'),
    formula = (FOCUS=='Economics') ~ poly(as.numeric(year), 2),
    family=binomial)
proportion_economic = function(year, subset=TRUE) {
  subset[is.na(subset)] = FALSE
  done = my_gs[subset & !is.na(my_gs$STATUS) & (my_gs$STATUS=='DONE'), ]
  denom = sum(done$year==year)
  numer = sum(done$year==year & done$FOCUS=='Economics')
  return(ifelse(denom==0, NA, numer/denom))
}
proportion_economic(year='2009',
              subset=my_gs$Oncotype.DX..Mammaprint.=='"Mammaprint only (70 genes)"')
plot(allyears, sapply(allyears, proportion_economic), xlab='year', ylab='proportion economic')
# plot(pch='m', col='red', allyears, 
#        sapply(allyears, proportion_economic,
#               subset=my_gs$Oncotype.DX..Mammaprint.=='"Mammaprint only (70 genes)"'), 
#        xlab='year', ylab='proportion economic')
```

